src/hotspot/share/utilities/compilerWarnings_gcc.hpp

```diff
@@ -63,4 +63,56 @@
 
 #endif // clang/gcc version check
 
+#if __GNUC__ >= 9
+
+// AIX also needs a 64 bit NULL to work as a null address pointer.
+// Most system includes on AIX would define it as an int 0 if not already defined with one
+// exception: /usr/include/dirent.h will unconditionally redefine NULL to int 0 again.
+// In this case you need to copy the following defines to a position after #include <dirent.h>
+#include <dirent.h>
+#ifdef _LP64
+  #undef NULL
+  #define NULL 0L
+#else
+  #ifndef NULL
+    #define NULL 0
+  #endif
+#endif
```

Comment:
This is really ugly just because we include dirent.h so we can add the warning for a few functions; and even uglier because it is only needed for AIX, and even uglier still because based on the existing code we only compile AIX with xlc - no? Otherwise we would already need this hack for gcc.

Reply:
Sorry, I'm confused. We build AIX with xlc. I don't believe we even include this file on AIX. How does this help AIX?

Reply:
I removed the changes for the dirent functions and removed the above code.  I also reverted all changes to os_aix.cpp.

Reply:
Thank you @hseigel !